### PR TITLE
chore: Remove deprecated ENABLE_ACCESS_REQUEST workflow

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

This PR partially addresses Superset 3.0 task [#9](https://github.com/orgs/apache/projects/201/views/1?pane=issue&itemId=17888789) by removing the previous deprecated `ENABLE_ACCESS_REQUEST` config parameter and the associated workflows.

The TL;DR is this legacy workflow failed to work post https://github.com/apache/superset/pull/22022 (which mentions that the API endpoint is deprecated and will be removed in 3.0) when the `GET` requests became `POST` requests meaning logic like [this](https://github.com/apache/superset/blob/9df8d8dda538095ba3b30d1f82aa81acf17e1c95/superset/views/core.py#L1903-L1905) and [this](https://github.com/apache/superset/blob/9df8d8dda538095ba3b30d1f82aa81acf17e1c95/superset/views/core.py#L902-L906) is currently broken.

Honestly I'm going in somewhat blind in terms of the how/where this logic was used, but the TL;DR is there _seems_ to be no approval logic/workflow if there's no request workflows. Hopefully I've trimmed enough dead code without over pruning, though this has turned somewhat into a yak shaving exercise as—by deleting the `/approve` endpoint—it led to removing facets of role extension, deleting the `granter` role, etc.

### TESTING INSTRUCTIONS

CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [x] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [x] Migration is atomic, supports rollback & is backwards-compatible
  - [x] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -888,21 +736,6 @@ def explore(
...
-
-        if datasource:
-            if config["ENABLE_ACCESS_REQUEST"] and (
-                not security_manager.can_access_datasource(datasource)
-            ):
-                flash(
-                    __(security_manager.get_datasource_access_error_msg(datasource)),
-                    "danger",
-                )
-                return redirect(
```
Comment (john-bodley) [inline superset/views/core.py:900]: Note this redirect no longer worked as this is being called  via a `GET` yet the redirect URL only accepts a `POST`.

**Review by dpgaspar — COMMENTED** [review]
It seems that probably makes sense to remove `/superset/override_role_permissions` also, since it seems related with `is granter` role. See if you agree and I'll close: https://github.com/apache/superset/pull/24334

**Comment by john-bodley** [comment]
@dpgaspar I agree with your comment as I went down this path previously. This also makes the `granter` role obsolete.

File: tests/integration_tests/datasets/api_tests.py [hunk 2]
```diff
@@ -348,7 +348,7 @@ def test_get_dataset_item(self):
             "sql": None,
             "table_name": "energy_usage",
             "template_params": None,
-            "uid": "2__table",
+            "uid": "6__table",
```
Comment (john-bodley) [inline tests/integration_tests/datasets/api_tests.py:351]: Love how removing some tests exposed the non-idempotent facets of others.

**Review by villebro — APPROVED** [review]
Great to see this dead code being removed, as this has caused a lot of confusion over the years. If this type of functionality really is needed, it should probably be done slightly differently, using pluggable FAB views/classes etc that don't require this much logic in the main codebase.

File: tests/integration_tests/datasets/api_tests.py (hunk 2 above)
Comment (john-bodley) [reply tests/integration_tests/datasets/api_tests.py:351]: I'll use `mock.ANY`.

**Comment by eschutho** [comment]
@john-bodley just to be on the safe side, I'm going to mark this as a possible breaking change. Do you think we should add anything to the UPDATING.md file re the changes?

**Pull request merged**